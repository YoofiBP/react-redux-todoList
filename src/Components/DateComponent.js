import React from 'react';
import IconButton from "@material-ui/core/IconButton";
import ArrowForwardIosIcon from '@material-ui/icons/ArrowForwardIos';
import ArrowBackIosIcon from '@material-ui/icons/ArrowBackIos';
import dayjs from 'dayjs'
import { changeDate } from '../redux/actions';
import { connect } from 'react-redux';
import { DATE_BACK, DATE_FORWARD } from '../redux/actionTypes';

function DateComponent(props){
    let today = (props.currentDate === dayjs().format("MM/DD/YYYY")) ? "Today" : dayjs(props.currentDate).format('ddd, MMM D, YYYY')
    return (
        <>
        <IconButton
        aria-label="delete"
        onClick={() => props.changeDate(DATE_BACK)}
        edge="end"
        className="p-2"
        size="small"
        id="back"
      >
        <ArrowBackIosIcon />
      </IconButton>
      <h1>{today}</h1>
      <IconButton
        aria-label="delete"
        onClick={() => props.changeDate(DATE_FORWARD)}
        edge="end"
        className="p-2"
        size='small'
        id="forward"
      >
        <ArrowForwardIosIcon />
      </IconButton>
      </>
    )
}

const mapStateToProps = (state) => {
  return {currentDate: state.date }
}

const mapDispatchToProps = dispatch => {
  return {
    changeDate: direction => {
      dispatch(changeDate(direction));
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(DateComponent);