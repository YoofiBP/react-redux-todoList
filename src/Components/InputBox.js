import React from 'react';
import { connect } from 'react-redux';
import { addItem } from '../redux/actions';
//import { users } from '../users';

class InputBox extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            value:""
        }
        this.handleChange = this.handleChange.bind(this)
        this.handleSubmit = this.handleSubmit.bind(this)
    }

    handleChange(event){
        this.setState({value: event.target.value})
    }

    handleSubmit(event){
        if(this.state.value.length > 1){
        this.props.addNewTodo(this.state.value, this.props.currentDate)
        this.setState({value:""})
        }
        event.preventDefault();
    }

    render(){
    return (
        <div className="input-group mb-2">
            <form className="input-group mb-2 w-100" onSubmit={this.handleSubmit}>
  <input id="title" onChange={this.handleChange} type="text" className="form-control" value={this.state.value} placeholder="New Item" aria-label="New Item" aria-describedby="button-addon2"/>
  <input type="submit" className="btn btn-outline-secondary" value="Add Item" />
  </form>
</div>
    )
    }
}

export class Welcome extends React.Component{
    constructor(props){
        super(props);
        this.state = {user: {}}
    }

    componentDidMount() {
    fetch('/users.json')
    .then(res => res.json())
    .then(users => { return users.filter(user => user.userId === this.props.userId)})
    .then(user => this.setState({user: user[0]}))
    .catch(err => console.log(err))
    }

    render() {
        return (
            <h1>Welcome {this.state.user.username}!</h1>
        )
    }
    
}

const mapDispatchToProps = dispatch => {
    return {
        addNewTodo: (title, date) => {
            dispatch(addItem(title, date))
        }
    }
}

const mapStateToProps = state => {
    return {
        currentDate: state.date
    }
}   

export default connect(mapStateToProps, mapDispatchToProps)(InputBox);