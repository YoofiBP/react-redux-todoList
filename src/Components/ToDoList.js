import React from 'react';
import TodoListItem from './ToDoListItem'
import { connect } from 'react-redux'

function ToDoList(props) {
    return (
        props.items.map(item => <TodoListItem item={item} key={item.id} id={item.id}/>)
    )
}

const mapStateToProps = state => {
    let [todaysItems] = state.todoList.filter(day => day.date === state.date)
    return todaysItems ? { items: todaysItems.tasks } : { items: [] }
}

export default connect(mapStateToProps)(ToDoList);