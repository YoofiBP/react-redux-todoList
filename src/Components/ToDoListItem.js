import React from "react";
import IconButton from "@material-ui/core/IconButton";
import DeleteIcon from "@material-ui/icons/Delete";
import InputIcon from "@material-ui/icons/Input";
import dayjs from "dayjs";
import Tooltip from "@material-ui/core/Tooltip";

import { connect } from 'react-redux';
import { addItem, checkItem, deleteItem } from '../redux/actions';

function TodoListItem(props) {
  const { title, completed } = props.item;
  return (
    <div className="d-flex">
      <li>
        <input
          type="checkbox"
          value={title}
          checked={completed}
          className="mr-auto p-2"
          onChange={() => props.checkItem(props.id,props.currentDate)}
        />{" "}
        <label
          className="mr-auto p-2"
          style={
            completed
              ? { textDecoration: "line-through", color: "grey" }
              : { textDecoration: "none" }
          }
        >
          {title}
        </label>
        <Tooltip title="Delete">
          <IconButton
            aria-label="delete"
            onClick={() => props.deleteItem(props.id, props.currentDate)}
            edge="end"
            className="p-2"
            id="delete"
          >
            <DeleteIcon />
          </IconButton>
        </Tooltip>
        <Tooltip title="Do Tomorrow">
          <IconButton onClick={() => props.moveItem(props.id, props.currentDate, title)} id="move">
            <InputIcon />
          </IconButton>
        </Tooltip>
      </li>
    </div>
  );
}

const mapDispatchToProps = dispatch => {
  return {
    checkItem: (index, date) => {
      dispatch(checkItem(index, date))
    },
    deleteItem: (index, date) => {
      dispatch(deleteItem(index, date))
    },
    moveItem: (index, date, title) => {
      dispatch(deleteItem(index, date));
      date = dayjs(date, "MM/DD/YYYY").add(1, "day").format("MM/DD/YYYY")
      dispatch(addItem(title, date))
    }
  }
}

const mapStateToProps = state => {
  return {
      currentDate: state.date
  }
} 
export default connect(mapStateToProps, mapDispatchToProps)(TodoListItem);