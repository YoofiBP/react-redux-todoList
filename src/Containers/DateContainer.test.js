import React from 'react';
import { DateComponent } from '../Components/DateComponent'
import { shallow } from 'enzyme';
import { Welcome } from './DateContainer'
import { create } from 'react-test-renderer';
import dayjs from "dayjs";

const testDate = dayjs("03/17/2020").format("MM/DD/YYYY")
const testUserId = 1;

describe('DateComponent Tests', () => {
    it('renders DateComponent without crashing', () => {
        shallow(<DateComponent />);
    })
    
    it('DateComponent matches snapshot', () => {
        const dateComponent = create(<DateComponent date={testDate}/>);
        expect(dateComponent.toJSON()).toMatchSnapshot();
    })  

    it('Renders today when today', () => {
        const date = dayjs().format("MM/DD/YYYY");
        const dateComponentWrapper = shallow(<DateComponent date={date}/>);
        expect(dateComponentWrapper.text()).toEqual("Today")
    })

    it('OnClick listeners working', () => {
        const dateBackward = jest.fn();
        const dateForward = jest.fn();
        const dateComponentWrapper = shallow(<DateComponent dateForward={dateForward} dateBackward={dateBackward}/>);
        const backbutton = dateComponentWrapper.find("#back");
        backbutton.simulate('click');
        expect(dateBackward).toHaveBeenCalled();
        const forwardbutton = dateComponentWrapper.find("#forward");
        forwardbutton.simulate('click');
        expect(dateForward).toHaveBeenCalled();
    })
})

describe('Welcome Tests', () => {
    it("Welcome component matches snapshot", () => {
        const welcome = create(<Welcome userId={testUserId}/>);
        expect(welcome.toJSON()).toMatchSnapshot()
    })

    it("Renders welcome text", () => {
        const welcomeWrapper = shallow(<Welcome userId={testUserId}/>);
        expect(welcomeWrapper.text()).toEqual("Welcome, Yoofi!")
    })
})

