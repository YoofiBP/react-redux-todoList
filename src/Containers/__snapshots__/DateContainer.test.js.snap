// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateComponent Tests DateComponent matches snapshot 1`] = `
Array [
  <button
    aria-label="delete"
    className="MuiButtonBase-root MuiIconButton-root p-2 MuiIconButton-sizeSmall MuiIconButton-edgeEnd"
    disabled={false}
    id="back"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>,
  <h1>
    Tue, Mar 17, 2020
  </h1>,
  <button
    aria-label="delete"
    className="MuiButtonBase-root MuiIconButton-root p-2 MuiIconButton-sizeSmall MuiIconButton-edgeEnd"
    disabled={false}
    id="forward"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>,
]
`;

exports[`DateContainer Tests Date Container matches snapshot 1`] = `
Array [
  <button
    aria-label="delete"
    className="MuiButtonBase-root MuiIconButton-root p-2 MuiIconButton-sizeSmall MuiIconButton-edgeEnd"
    disabled={false}
    id="back"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
        />
      </svg>
    </span>
  </button>,
  <h1>
    Tue, Mar 17, 2020
  </h1>,
  <button
    aria-label="delete"
    className="MuiButtonBase-root MuiIconButton-root p-2 MuiIconButton-sizeSmall MuiIconButton-edgeEnd"
    disabled={false}
    id="forward"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
        />
      </svg>
    </span>
  </button>,
]
`;

exports[`Welcome Tests Welcome component matches snapshot 1`] = `
<h1>
  Welcome, 
  Yoofi
  !
</h1>
`;
